lua/controller.lua

```diff
@@ -273,11 +274,18 @@ local function controller_insert_to_drawers(pos, stack)
 		-- store, the drawer is not full, and the drawer entity is loaded, we
 		-- will put the items in the drawer
 		if content.name == stack:get_name() and
-				content.count < content.maxCount and
 				drawers.drawer_visuals[core.hash_node_position(drawer_pos)] then
-			return drawers.drawer_insert_object(drawer_pos, stack, visualid)
+			if content.count < content.maxCount then
+				return drawers.drawer_insert_object(drawer_pos, stack, visualid)
+			else
+				new_drawer = true
+			end
 		end
-	elseif drawer_net_index["empty"] then
+	else
+		new_drawer = true
+	end
+
+	if new_drawer and drawer_net_index["empty"] then
```

Comment:
Ok, I had considered taking this conditional block and re-factoring it out into a new function. However, I thought that would add unwanted overhead performing identical setup and add risk of regressions.
My goal was minimum necessary code changes, but I'll look again when I have the time.